ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/services/commands/tracker/CompletionResponse.scala

```diff
@@ -92,22 +93,14 @@ object CompletionResponse {
   private[daml] def toException(response: TrackedCompletionFailure): StatusException =
     response match {
       case QueueCompletionFailure(failure) =>
-        toException(failure)
+        val metadata = failure.metadata
```

Comment:
This is the only place we use `CompletionFailure#metadata`. I understand that it's necessary here but we can probably remove it from the trait and the other subtypes.